core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -328,4 +365,70 @@ public ShareFetchSessionCache(int maxEntries, long evictionMs) {
             this.evictionMs = evictionMs;
         }
     }
+
+    /*
+     * A cached partition.
+     *
+     * The broker maintains a set of these objects for each incremental share fetch session.
+     * When an incremental share fetch request is made, any partitions which are not explicitly
+     * enumerated in the fetch request are loaded from the cache.  Similarly, when an
+     * incremental share fetch response is being prepared, any partitions that have not changed and
+     * do not have errors are left out of the response.
+     *
+     * We store many of these objects, so it is important for them to be memory-efficient.
+     * That is why we store topic and partition separately rather than storing a TopicPartition
+     * object. The TP object takes up more memory because it is a separate JVM object, and
+     * because it stores the cached hash code in memory.
+     *
+     */
+    public static class CachedPartition implements ImplicitLinkedHashCollection.Element {
```

Comment:
How is it different than SharePartiton? Isn't SharePartition our cached object? Is it more like Session cached Partition which internally should have SharePartition as latter should work without session/next session as well?

Reply:
I don't see this being used as well.

Reply:
Yes, you're right. This is a session cached partition which I haven't used so far in the code but will be utilized whenever we used a share session based fetch